### PR TITLE
Bjc/usb fixes

### PR DESCRIPTION
**Pull request by bjc**
This covers more bases when the device is reset by hooking into the firmware library's reset hook.

I've also added a 5ms transmission timeout on flush so things don't hang indefinitely if the host is ignoring that endpoint.

Tested on macOS UEFI and the real OS, as well as Linux. I don't have access to windows right now, so I can't test that.

### CODE REVIEW

File: cores/arduino/USBCore.cpp [hunk 1]
```diff
@@ -431,6 +448,14 @@ class ClassCore
         }
 };
 
+void (*oldResetHandler)(usb_dev *usbd);
+void handleReset(usb_dev *usbd)
+{
+    Serial1.print("/");
```
Comment (maxgerhardt) [inline cores/arduino/USBCore.cpp:454]: Is this a debugging leftover that prints something to the hardware UART?

Comment (bjc) [reply cores/arduino/USBCore.cpp:454]: oops! give me a tick to push a fix

**Comment by maxgerhardt** [comment]
I'll test this shortly on Windows and merge if USB-CDC still works as normal.

**Comment by maxgerhardt** [comment]
Sorry for the wait, I'll test shortly.

**Comment by maxgerhardt** [comment]
Sorry for the massive delay, I've tested this to be *working* on Windows, thus I'll merge.

grafik

**Pull request merged**